packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -528,6 +630,49 @@ export class StringListParameter extends ParameterBase implements IStringListPar
     return new Import(scope, id);
   }
 
+  /**
+   * Imports an external string list parameter with name and optional version.
+   */
+  public static fromListParameterAttributes(scope: Construct, id: string, attrs: ListParameterAttributes): IStringListParameter {
+    if (!attrs.parameterName) {
+      throw new Error('parameterName cannot be an empty string');
+    }
+
+    const type = attrs.valueType ?? ParameterValueType.STRING;
+    const valueType = `List<${type}>`;
+
+    const stringValue = attrs.version
+      ? new CfnDynamicReference(CfnDynamicReferenceService.SSM, `${attrs.parameterName}:${Tokenization.stringifyNumber(attrs.version)}`).toStringList()
+      : new CfnParameter(scope, `${id}.Parameter`, { type: `AWS::SSM::Parameter::Value<${valueType}>`, default: attrs.parameterName }).valueAsList;
+
+    class Import extends ParameterBase {
+      public readonly parameterName = attrs.parameterName;
+      public readonly parameterArn = arnForParameterName(this, attrs.parameterName, { simpleName: attrs.simpleName });
+      public readonly parameterType = valueType; // it doesn't really matter what this is since a CfnParameter can only be `String | StringList`
+      public readonly stringListValue = stringValue;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Returns a token that will resolve (during deployment) to the list value of an SSM StringList parameter.
+   * @param scope Some scope within a stack
+   * @param parameterName The name of the SSM parameter.
+   * @param type the type of the SSM list parameter
+   * @param version The parameter version (recommended in order to ensure that the value won't change during deployment)
+   */
+  public static valueForTypedListParameter(scope: Construct, parameterName: string, type = ParameterValueType.STRING, version?: number): string[] {
```

Comment:
Does typing this value even add value? (ha)

We return it as a `string[]` anyway... so does anybody does any checking on the types at all? If so, who?

Reply:
CloudFormation does the type checking in this case. If you request a `ParameterValueType.AWS_EC2_IMAGE_ID` it expects the value to be a string in the AMI format, and if it is not it will throw an error.